src/rgw/rgw_rest_s3.cc

```diff
@@ -1441,6 +1441,12 @@ int RGWSetBucketWebsite_ObjStore_S3::get_params()
                      + ".";
     return -ERR_INVALID_REQUEST;
   }
+  for (list<RGWBWRoutingRule>::iterator iter = website_conf.routing_rules.rules.begin(); iter != website_conf.routing_rules.rules.end(); ++iter) {
```

Comment:
lets use a ranged-for loop here: `for (const auto& rule : website_conf.routing_rules.rules) {`